### ISSUE TITLE
Datagrid columns header and screen reader announcement accessibility issue 

### ISSUE THREAD

**Issue by pavandixit93** [vmware/clarity#6010]
## Describe the bug

Clarity datagrid column header has role `columnheader`, due to this role when i focus on datagrid header first time, JAWS is announcing lengthy message. I found that JAWS is announcing that `Column resize handle` and associated `aria-describedby` message if I am focusing on column header button first time.

## How to reproduce

1.  Go to 'https://v4.clarity.design/documentation/datagrid/custom-sorting'
2.  Open screen reader `Jaws` 
3.  Press `t` so that JAWS will focus the table/datagrid. 
4.  Press `CTRL+ALT+LEFT arrow key` (for windows) so that it will navigate to each column header, On every column header is it announcing lengthy message, Also for sure please check for `Pokemon` column header, it announces such lengthy message.
5. Repeat steps 1 to 3 and now press `TAB` key and focus `Pokemon` column header and it announces combined aria-label of descendent elements. 
6. This happens if i press `TAB` or `table key shortcuts` for column navigation, If i navigate in reverse i.e. by pressing `SHIFT+TAB` JAWS announces the focused elements `aria-label` only.

## Expected behavior

I agree that due to role `columnheader` JAWS announces such lengthy message but when i use `TAB` key to focus each column header, i can focus the `Column resize handle` button so screen reader should announce the focused element's `aria-label` and not the combined aria-label of descendent elements. 

Expected Message: `Pokemon button sorted ascending`
Actual Message: `Pokemon Toggle column filter Column resize handle Not necessary to use this button` 

As shown in below image, please focus the `Pokemon` column.
![image](https://user-images.githubusercontent.com/21095558/120146370-962a2680-c202-11eb-828c-6191c210c175.png)

## Versions

**Clarity project:**

- [ ] Clarity Core
- [ ] Clarity Angular/UI

**Clarity version:**

- [ ] v3.x
- [ ] v4.x
- [ ] v5.x

**Framework:**

- [ ] Angular
- [ ] React
- [ ] Vue
- [ ] Other:

**Framework version:**
_ie: Angular 11_

**Device:**

- Type: [e.g. MacBook]

## Additional notes

_Add any other notes about the problem here._

**Comment by Jinnie** [comment]
Need A11Y team comment for this issue.
@scroniser is this a problem, or it's working as expected?

**Comment by scroniser** [comment]
@Jinnie - Confirming this is an issue, though it's not been filed under this specific topic. This is a violation of WCAG 2.1 2.4.6: Headings and Labels (Level AA). Please let me know if you'd like to discuss further.

**Comment by scroniser** [comment]
Added this issue to the internal tracking system.

**Comment by hippee-lee** [comment]
Thanks @scroniser - I'll close this as it has been triaged and is being tracked with the other a11y issues. 

Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.

**Comment by pavandixit93** [comment]
Thank you team!!
